### PR TITLE
retry failed migrations one time

### LINKED ISSUE #9: Retry failed exports
Sometimes, a large portion of the started migrations fails. These should be retried a  few times.

### PR DESCRIPTION
**Pull request by soofstad**
Some logic to retry the list of failed migrations.

closes #9

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -136,3 +136,4 @@ dmypy.json
 /bin/
 /obj/
 /tmp/
+out
```
Comment (sindre-nistad) [inline .gitignore]: Newline?

File: .env-template [hunk 2]
```diff
@@ -1,2 +1,5 @@
 GITHUB_TOKEN=
-STORAGE_KEY=
+STORAGE_KEY="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1;"
+WEEKLY_RETENTION=1
+GITHUB_ORG=equinor
+BLOB_CONTAINER=github-archives
```
Comment (sindre-nistad) [inline .env-template]: Good idea to put them in `.env`.

Maybe add a comment over `STORAGE_KEY`, that it is the default secret, and add a link to where one can find more information (same URL as in `decker-compose.override.yml`).
That way, a new developer won't be surprised / shocked to find a hard-coded secret.

File: docker-compose.override.yml [hunk 3]
```diff
@@ -13,20 +13,11 @@ services:
...
     image: mcr.microsoft.com/azure-storage/azurite
     volumes:
-      - "blobStorage:/data"
+      - "./blobStorage:/data"
     ports:
-      - "10000:10000"  # Blobs
-    command: [
-        "azurite",
-        "--blobHost", "0.0.0.0",
-        "--oauth", "basic",
```
Comment (sindre-nistad) [inline docker-compose.override.yml:28]: `--oauth basic` may be necessary for azurite to work properly. Did you test it without, and confirm that it works as expected?

File: .gitignore (hunk 1 above)
Comment (sindre-nistad) [reply .gitignore]: We should add `blobStorage` as well, due to the change in volumes in `docker-compose.override.yml`

File: docker-compose.override.yml [hunk 4]
```diff
@@ -13,20 +13,11 @@ services:
...
     depends_on:
       - storage-emulator
-    env_file:
-      - .env
 
   storage-emulator:
     image: mcr.microsoft.com/azure-storage/azurite
     volumes:
-      - "blobStorage:/data"
+      - "./blobStorage:/data"
```
Comment (sindre-nistad) [inline docker-compose.override.yml:11]: Any particular reason to store the persistent data to the local folder?

File: src/Program.cs [hunk 5]
```diff
@@ -8,99 +9,115 @@ namespace ms_continuus
...
 
-            var blobStorage = new BlobStorage();
-            await blobStorage.EnsureContainer();
-            await blobStorage.DeleteArchivesBefore(olderThan, "monthly");
+        private static async Task<bool> DownloadAndUpload(Migration migration, int index)
+        {
+            Migration migStatus = await Api.MigrationStatus(migration.Id);
+            var exportTimer = 0;
+            const int sleepIntervalSeconds = 30;
+            while (migStatus.State != "exported")
```
Comment (sindre-nistad) [inline src/Program.cs]: Do we have an enum, instead of comparing with a string?

File: docker-compose.override.yml (hunk 4 above)
Comment (soofstad) [reply docker-compose.override.yml:11]: Personal preference. 
I like having the actual data easily available for browsing, backup and cleanup.
Without having to go sudo into some "/var/lib/docker/volume" path, that may change on different systems.

File: docker-compose.override.yml (hunk 3 above)
Comment (soofstad) [reply docker-compose.override.yml:28]: Tested ok.
Following this doc; https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite#oauth-configuration
I see no reason why we should enable the flag.

Comment (sindre-nistad) [reply docker-compose.override.yml:28]: Ok, it seemed to be necessary, when I was figuring out how to set it up.

File: docker-compose.override.yml (hunk 4 above)
Comment (sindre-nistad) [reply docker-compose.override.yml:11]: Fair enough, just remember to add the folder in `.gitignore`.

File: .gitignore (hunk 1 above)
Comment (soofstad) [reply .gitignore]: fixed

File: .env-template (hunk 2 above)
Comment (soofstad) [reply .env-template]: fixed

File: src/Program.cs (hunk 5 above)
Comment (soofstad) [reply src/Program.cs]: fixed

**Review by sindre-nistad — APPROVED** [review]

**Review by sindre-nistad — APPROVED** [review]

**Pull request merged**